### PR TITLE
Feature/material properties UI cu e753kh

### PR DESCRIPTION
**Pull request by Mefiso**

### CODE REVIEW

**Review by JaimeBea — APPROVED** [review]

File: Project/Source/Resources/ResourceShader.h [hunk 1]
```diff
@@ -11,11 +15,16 @@ class ResourceShader : public Resource {
...
+	void SaveShaderType() const;
+
 	unsigned int GetShaderProgram() {
 		return shaderProgram;
 	}
 	ShaderType GetShaderType() {
-		return shaderType; 
+		return shaderType;
+	}
+	void SetShaderType(const ShaderType type) {
```
Comment (lowysole) [inline Project/Source/Resources/ResourceShader.h:26]: Implement Getter and Setter in .cpp

File: Project/Source/Components/ComponentMeshRenderer.cpp [hunk 2]
```diff
@@ -132,11 +169,10 @@ void ComponentMeshRenderer::OnEditorUpdate() {
 					} else {
 						ImGui::ResourceSlot<ResourceTexture>("Specular Texture", &material->specularMapId);
 					}
-
-					// Smoothness Combo
-					const char* smoothnessItems[] = {"Smoothness Value", "Smoothness Alpha"};
+					// Shininess Combo
```
Comment (lowysole) [inline Project/Source/Components/ComponentMeshRenderer.cpp:172]: Rename _smoothness_ variables to _shininess_ for PHONG case

File: Project/Source/Components/ComponentMeshRenderer.cpp [hunk 3]
```diff
@@ -295,128 +419,132 @@ void ComponentMeshRenderer::Draw(const float4x4& modelMatrix) const {
...
+		unsigned glTextureSpecular = 0;
+		ResourceTexture* specular = (ResourceTexture*) App->resources->GetResource(material->specularMapId);
+		glTextureSpecular = specular ? specular->glTexture : 0;
+
+		// Phong-specific uniform settings
 		glUseProgram(program);
 
-		glUniform3fv(glGetUniformLocation(program, "diffuseColor"), 1, material->diffuseColor.ptr());
 		glUniform3fv(glGetUniformLocation(program, "specularColor"), 1, material->specularColor.ptr());
 		int hasDiffuseMap = diffuse ? 1 : 0;
```
Comment (lowysole) [inline Project/Source/Components/ComponentMeshRenderer.cpp:549]: material->hasDiffuseMap

File: Project/Source/Components/ComponentMeshRenderer.cpp [hunk 4]
```diff
@@ -295,128 +419,132 @@ void ComponentMeshRenderer::Draw(const float4x4& modelMatrix) const {
...
 		int hasSpecularMap = specular ? 1 : 0;
```
Comment (lowysole) [inline Project/Source/Components/ComponentMeshRenderer.cpp:550]: material->hasSpecularMap

File: Project/Source/Resources/ResourceShader.cpp [hunk 5]
```diff
@@ -2,7 +2,50 @@
...
+#include "FileSystem/JsonValue.h"
 #include "Modules/ModulePrograms.h"
+#include "Modules/ModuleFiles.h"
+#include "Modules/ModuleTime.h"
+
+#include "rapidjson/error/en.h"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/prettywriter.h"
+#include "rapidjson/document.h"
+
```
Comment (Jordi-Gil) [inline Project/Source/Resources/ResourceShader.cpp:14]: Falta el Leaks.h

**Review by Jordi-Gil — APPROVED** [review]

**Review by lowysole — APPROVED** [review]

**Pull request merged**